lua/starfall/libs_sh/http.lua

```diff
@@ -23,19 +23,43 @@ local checkpermission = instance.player ~= SF.Superuser and SF.Permissions.check
 
 local http_library = instance.Libraries.http
 
+local function confirmPlayerHasRequests(player)
+	if not requests[player] then
+		requests[player] = 0
+	end
+end
+local function addToPlayerRequests(player, val)
+	confirmPlayerHasRequests(player)
+	requests[player] = math.Clamp(requests[player] + val, 0, http_max_active:GetInt())
+end
 -- Runs the appropriate callback after a http request
 local function runCallback(callback)
 	return function(...)
 		if callback then
+			addToPlayerRequests(instance.player, -1)
```

Comment:
I think this should be outside of the `if callback then` block since the user can make requests without a fail callback